src/KOKKOS/fix_rigid_kokkos.cpp

```diff
@@ -0,0 +1,1274 @@
+{
+  // TODO: PORT THIS; CONTINUE HERE LATER.
+  int i,ibody;
+  
+  // sum over atoms to get force and torque on rigid body
+
+  double **x = atom->x;
+  double **f = atom->f;
+  int nlocal = atom->nlocal;
+
+  double dx,dy,dz;
+  double unwrap[3];
+
+  for (ibody = 0; ibody < nbody; ibody++)
+    for (i = 0; i < 6; i++) sum[ibody][i] = 0.0;
+
+  for (i = 0; i < nlocal; i++) {
+    if (body[i] < 0) continue;
+    ibody = body[i];
+
+    sum[ibody][0] += f[i][0];
+    sum[ibody][1] += f[i][1];
+    sum[ibody][2] += f[i][2];
+
+    domain->unmap(x[i],xcmimage[i],unwrap);
+    dx = unwrap[0] - xcm[ibody][0];
+    dy = unwrap[1] - xcm[ibody][1];
+    dz = unwrap[2] - xcm[ibody][2];
+
+    sum[ibody][3] += dy*f[i][2] - dz*f[i][1];
+    sum[ibody][4] += dz*f[i][0] - dx*f[i][2];
+    sum[ibody][5] += dx*f[i][1] - dy*f[i][0];
+  }
+
+  // extended particles add their torque to torque of body
+
+  if (extended) {
+    double **torque_one = atom->torque;
+
+    for (i = 0; i < nlocal; i++) {
+      if (body[i] < 0) continue;
+      ibody = body[i];
+
+      if (eflags[i] & TORQUE) {
+        sum[ibody][3] += torque_one[i][0];
+        sum[ibody][4] += torque_one[i][1];
+        sum[ibody][5] += torque_one[i][2];
+      }
+    }
+  }
+
+  MPI_Allreduce(sum[0],all[0],6*nbody,MPI_DOUBLE,MPI_SUM,world);
+
+  // include Langevin thermostat forces
+
+  for (ibody = 0; ibody < nbody; ibody++) {
+    fcm[ibody][0] = all[ibody][0] + langextra[ibody][0];
+    fcm[ibody][1] = all[ibody][1] + langextra[ibody][1];
+    fcm[ibody][2] = all[ibody][2] + langextra[ibody][2];
+    torque[ibody][0] = all[ibody][3] + langextra[ibody][3];
+    torque[ibody][1] = all[ibody][4] + langextra[ibody][4];
+    torque[ibody][2] = all[ibody][5] + langextra[ibody][5];
+  }
+
+}
+
+
+template <class DeviceType>
+void FixRigidKokkos<DeviceType>::set_v_kokkos()
+{
+  double xprd = domain->xprd;
+  double yprd = domain->yprd;
+  double zprd = domain->zprd;
+
+  int nlocal = atomKK->nlocal;
+
+  x = atomKK->k_x.view<DeviceType>();
+  v = atomKK->k_v.view<DeviceType>();
+  f = atomKK->k_f.view<DeviceType>();
+  rmass = atomKK->k_rmass.view<DeviceType>();
+  mass = atomKK->k_mass.view<DeviceType>();
+  type = atomKK->k_type.view<DeviceType>();
+  mask = atomKK->k_mask.view<DeviceType>();
+
+  double xy, xz,yz;
+
+  if (triclinic) {
+    xy = domain->xy;
+    xz = domain->xz;
+    yz = domain->yz;
+  }
+
+  // set v of each atom
+  {
+    // Local block for Kokkos parallel for:
+    
+    auto l_x = x;
+    auto l_v = v;
+    auto l_f = f;
+
+    auto l_type = type;
+    auto l_rmass = rmass;
+    auto l_mass  = mass;
+    
+    auto l_ex_space = k_ex_space.d_view;
+    auto l_ey_space = k_ex_space.d_view;
+    auto l_ez_space = k_ex_space.d_view;
+
+    auto l_vcm = k_vcm.d_view;
+    auto l_omega = k_omega.d_view;
+    auto l_displace = k_displace.d_view;
+    auto l_body = k_body.d_view;
+    
+    Kokkos::parallel_for(nlocal, LAMMPS_LAMBDA(const int& i) {
```

Comment:
Needs to be a `Kokkos::parallel_reduce` for energy and virial calcs

Reply:
I might need a hand with this or a pointer to another piece of code that does what needs to be done.

Reply:
For example, see https://github.com/lammps/lammps/blob/master/src/KOKKOS/pair_coul_wolf_kokkos.cpp#L127 and https://github.com/lammps/lammps/blob/master/src/KOKKOS/pair_coul_wolf_kokkos.cpp#L296